src/rtp.c

```diff
@@ -1020,11 +1020,18 @@ void janus_rtp_simulcasting_context_reset(janus_rtp_simulcasting_context *contex
 void janus_rtp_simulcasting_prepare(json_t *simulcast, int *rid_ext_id, uint32_t *ssrcs, char **rids) {
 	if(simulcast == NULL)
 		return;
+	/* Clear existing RIDs in case this array was reused from a previous call */
+	size_t i = 0;
+	for(i=0; i<3; i++) {
```

Comment:
Your new commit made me realize that you should have an `if(rids)` for this block as well, since it may be `NULL`.

Reply:
Ah, right, sorry, good catch.